docs/reference/operations/create_transfers.md

```diff
@@ -0,0 +1,309 @@
+
+### `code_must_not_be_zero`
+The transfer was not created.
+[`Transfer.code`](../transfers.md#code) is zero, but must be nonzero.
+
+### `amount_must_not_be_zero`
+The transfer was not created.
+[`Transfer.amount`](../transfers.md#amount) is zero, but must be nonzero.
+
+Every transfer must move value.
+Only posting and voiding transfer amounts may be zero — when zero, they will move the full pending
+amount.
+
+### `debit_account_not_found`
+The transfer was not created.
+[`Transfer.debit_account_id`](../transfers.md#debit_account_id) must refer to an existing `Account`.
+
+### `credit_account_not_found`
+The transfer was not created.
+[`Transfer.credit_account_id`](../transfers.md#credit_account_id) must refer to an existing `Account`.
+
+### `accounts_must_have_the_same_ledger`
+The transfer was not created. The accounts referred to by
+[`Transfer.debit_account_id`](../transfers.md#debit_account_id) and
+[`Transfer.credit_account_id`](../transfers.md#credit_account_id) must have an identical
+[`ledger`](../accounts.md#ledger).
+
+(TODO: Link to explanation of implementing currency exchange with liquidity accounts).
+
+### `transfer_must_have_the_same_ledger_as_accounts`
+The transfer was not created. The accounts referred to by
+[`Transfer.debit_account_id`](../transfers.md#debit_account_id) and
+[`Transfer.credit_account_id`](../transfers.md#credit_account_id) are equivalent, but
+differ from the [`Transfer.ledger`](../transfers.md#ledger).
+
+### `exists_with_different_flags`
+The transfer already exists, but with different [`flags`](../transfers.md#flags).
+
+### `exists_with_different_debit_account_id`
+The transfer already exists, but with a different [`debit_account_id`](../transfers.md#debit_account_id).
+
+### `exists_with_different_credit_account_id`
+The transfer already exists, but with a different [`credit_account_id`](../transfers.md#credit_account_id).
+
+### `exists_with_different_user_data`
+The transfer already exists, but with a different [`user_data`](../transfers.md#user_data).
+
+### `exists_with_different_pending_id`
+The transfer already exists, but with a different [`pending_id`](../transfers.md#pending_id).
+
+### `exists_with_different_timeout`
+The transfer already exists, but with a different [`timeout`](../transfers.md#timeout).
+
+### `exists_with_different_code`
+The transfer already exists, but with a different [`code`](../transfers.md#code).
+
+### `exists_with_different_amount`
+The transfer already exists, but with a different [`amount`](../transfers.md#amount).
+
+### `exists`
+The transfer already exists, and is identical to the transfer in the request.
+
+To correctly recover from application crashes
+[many applications](../../usage/integration.md#conistency-with-foreign-databases) should handle
+`exists` exactly as [`ok`](#ok).
+
+### `overflows_debits_pending`
+The transfer was not created.
+`debit_account.debits_pending + transfer.amount` would overflow a 64-bit unsigned integer.
```

Comment:
Not that we necessarily need to cover it in this PR but what is supposed to happen in business logic in this case?

Reply:
u64's are very big... if they hit this limit it is probably an application bug, or their data model was designed wrong (i.e. they should have used a different asset scale). Maybe in some cryptocurrency use cases, where there are extremely high-precision currencies, you may have to have multiple `ledger`s for the same currency, where they are different scales. (e.g. store "cents" and "dollars" in separate accounts, with different scales). That would be extra work for the application, but it is an extremely niche use case.